Wrappers/Python/ccpi/optimisation/operators/GradientOperator.py

```diff
@@ -39,29 +60,80 @@ class Gradient(LinearOperator):
                 Example (2D): u\in X, \nabla(u) = [\partial_{y} u, \partial_{x} u]
                               u^{*}\in Y, \nabla^{*}(u^{*}) = \partial_{y} v1 + \partial_{x} v2
         
-                Grad_order = ['channels', 'direction_z', 'direction_y', 'direction_x']
-                Grad_order = ['channels', 'direction_y', 'direction_x']
-                Grad_order = ['direction_z', 'direction_y', 'direction_x']
-                Grad_order = ['channels', 'direction_z', 'direction_y', 'direction_x']
-                
-
     '''
+
+    #kept here for backwards compatability
+    CORRELATION_SPACE = CORRELATION_SPACE
+    CORRELATION_SPACECHANNEL = CORRELATION_SPACECHANNEL
+
+    def __init__(self, gm_domain, bnd_cond = 'Neumann', **kwargs):
+        """Constructor method
+        """        
+        super(Gradient, self).__init__() 
+
+        backend = kwargs.get('backend',C)
+
+        correlation = kwargs.get('correlation',CORRELATION_SPACE)
+
+        if correlation == CORRELATION_SPACE and gm_domain.channels > 1:
+            #numpy implementation only for now
+            backend = NUMPY
+            "Warning: correlation='Space' on multi-channel dataset will use `numpy` backend"
+
+        if backend == NUMPY:
+            self.Operator = Gradient_numpy(gm_domain, bnd_cond=bnd_cond, **kwargs)
```

Comment:
I think we decided to stick to [SIRF](https://github.com/CCPPETMR/SIRF/blob/master/doc/DeveloperGuide.md#conventions_naming)'s coding convention and instances are lower case: `self.operator` instead of `self.Operator`.